File: 2018_01__Python_introduction_for_MATLAB_users__at_Supelec/examples/1D_numerical_integration_and_plot.py
#! /usr/bin/env python
# -*- coding: utf8 -*-

import numpy as np                # standard convention
import matplotlib.pyplot as plt   # standard convention
from scipy.integrate import quad  # need only 1 function


def Ei(x, minfloat=1e-6, maxfloat=1000):
    def f(t):
        return np.exp(-t) / t
    if x > 0:
        return -1.0 * (quad(f, -x, -minfloat)[0]
                       + quad(f, minfloat, maxfloat)[0])
    else:
        return -1.0 * quad(f, -x, maxfloat)[0]


X = np.linspace(-1, 1, 1000)  # 1000 points
Y = np.vectorize(Ei)(X)       # or [Ei(x) for x in X]
plt.plot(X, Y)                # MATLAB-like interface
plt.title("The function Ei(x)")
plt.xlabel("x")
plt.ylabel("y")
plt.savefig("figures/Ei_integral.png")
plt.show()

File: 2018_06__Julia_my_new_optimization_friend__introduction_for_MATLAB_users__at_IETR_seminar/examples/1D_numerical_integration_and_plot.jl
#! /usr/bin/env julia

using QuadGK

function Ei(x, minfloat=1e-3, maxfloat=100)
    f = t -> exp(-t) / t  # inline function
    if x > 0
        return quadgk(f, -x, -minfloat)[1]
             + quadgk(f, minfloat, maxfloat)[1]
    else
        return quadgk(f, -x, maxfloat)[1]
    end
end

X = linspace(-1, 1, 1000)   # 1000 points
Y = [ Ei(x) for x in X ]

using Winston
plot(X, Y)
title("The function Ei(x)")
xlabel("x"); ylabel("y")
savefig("figures/Ei_integral.png")
